File: core/evidence/semantic_truth_limit_engine.py
from __future__ import annotations

from typing import Any, Dict


def semantic_truth_limits(entropy: Dict[str, Any], instability: Dict[str, Any]) -> Dict[str, Any]:
    return {
        "stabilization_allowed": not entropy.get("suppress_stabilization", False),
        "coherence_allowed": not entropy.get("suppress_coherence", False),
        "instability_preserved": instability.get("preserved", True),
    }

File: src/evidence/semanticTruthLimitEngine.ts
/**
 * Converted from Python: core/evidence/semantic_truth_limit_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function semanticTruthLimits(entropy: any, instability: any): any {
  return {"stabilization_allowed": !py.truthy(py.get(entropy, "suppress_stabilization", false)), "coherence_allowed": !py.truthy(py.get(entropy, "suppress_coherence", false)), "instability_preserved": py.get(instability, "preserved", true)};
}
